### PR TITLE
📝(README) update readme with instructions about xapi-service image

### PR DESCRIPTION
**Pull request by lunika**
## Purpose

Readme was not up-to-date with instructions about xapi-service image.
Instructions about how to tag image is also added.

## Proposal

- [x] update readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,23 +1,46 @@
 # Learning locker docker images
 
-This repository contains the `Dockerfile` to build images for [learning locker](https://github.com/LearningLocker/learninglocker), a LRS to store and analyze xAPI statements.
+This repository contains the `Dockerfile` to build images for:
 
-We follow the learning locker's tag, so an image will have the same tag you can found in learning locker repository.
+- [learning locker](https://github.com/LearningLocker/learninglocker), a LRS to store and analyze xAPI statements.
+- [xapi-service](https://github.com/LearningLocker/xapi-service), a service for xAPI develop by learning locker.
```
Comment (jmaupetit) [inline README.md]:
```suggestion
- [xapi-service](https://github.com/LearningLocker/xapi-service), a service dedicated to xAPI requests.
```

File: README.md [hunk 2]
```diff
@@ -1,23 +1,46 @@
...
+
+We follow the learning locker's tag, so an image will have the same tag you can found in learning locker and xapi-service repositories.
```
Comment (jmaupetit) [inline README.md]:
```suggestion
We follow learning locker's versions, so images built from this repository will have the same tags you can found in learning locker and xapi-service repositories.
```

File: README.md [hunk 3]
```diff
@@ -1,23 +1,46 @@
...
 
 ## Building a learning locker image
 
-To build a learning locker image, you have to pick the right tag to use and then build it using the docker command, Here with the tag `v2.6.2:
+To build a learning locker image, you have to pick the right [tag](https://github.com/LearningLocker/learninglocker/tags) to use and then build it using the docker command, Here with the tag `v2.6.2`:
```
Comment (jmaupetit) [inline README.md]:
```suggestion
To build a learning locker image, you have to pick the right [tag](https://github.com/LearningLocker/learninglocker/tags) to use and then build it using the following docker command (here with the `v2.6.2` tag):
```

File: README.md [hunk 4]
```diff
@@ -1,23 +1,46 @@
...
+
+```bash
+$ docker build -t fundocker/learninglocker:v2.6.2 --build-arg LL_VERSION="v2.6.2" learning/
+```
+
+## Building a xapi-service image
```
Comment (jmaupetit) [inline README.md]:
```suggestion
## Building xapi-service image
```

File: README.md [hunk 5]
```diff
@@ -1,23 +1,46 @@
...
-This repository contains the `Dockerfile` to build images for [learning locker](https://github.com/LearningLocker/learninglocker), a LRS to store and analyze xAPI statements.
+This repository contains the `Dockerfile` to build images for:
 
-We follow the learning locker's tag, so an image will have the same tag you can found in learning locker repository.
+- [learning locker](https://github.com/LearningLocker/learninglocker), a LRS to store and analyze xAPI statements.
+- [xapi-service](https://github.com/LearningLocker/xapi-service), a service for xAPI develop by learning locker.
+
+We follow the learning locker's tag, so an image will have the same tag you can found in learning locker and xapi-service repositories.
 
 ## Building a learning locker image
```
Comment (jmaupetit) [inline README.md]:
```suggestion
## Building learning locker image
```

File: README.md [hunk 6]
```diff
@@ -1,23 +1,46 @@
...
-To build a learning locker image, you have to pick the right tag to use and then build it using the docker command, Here with the tag `v2.6.2:
+To build a learning locker image, you have to pick the right [tag](https://github.com/LearningLocker/learninglocker/tags) to use and then build it using the docker command, Here with the tag `v2.6.2`:
+
+```bash
+$ docker build -t fundocker/learninglocker:v2.6.2 --build-arg LL_VERSION="v2.6.2" learning/
+```
+
+## Building a xapi-service image
+
+To build a xapi-service image, you have to pick the right [tag](https://github.com/LearningLocker/xapi-service/tags) to use and then build it using the docker command, Here with the tag `v2.2.15`:
```
Comment (jmaupetit) [inline README.md]:
```suggestion
To build the xapi-service image, you have to pick the right [tag](https://github.com/LearningLocker/xapi-service/tags) to use and then build it using the following docker command (here with the `v2.2.15` tag):
```

File: README.md [hunk 7]
```diff
@@ -1,23 +1,46 @@
...
 
 ```bash
-$ docker build -t fundocker/learninglocker:v2.6.2 --build-arg LL_VERSION="v2.6.2" .
+$ docker build -t fundocker/xapi-service:v2.2.15 --build-arg VERSION="v2.2.15" xapi/
 ```
 
+## How to use the CI
+
+The CI configured with this repository will test if the image can be built correctly but is also able to push images to docker hub when we tag this repository.
```
Comment (jmaupetit) [inline README.md]:
```suggestion
The CI configured in this repository will test if the image can be built successfully for a particular release, but it also pushes images to docker hub when this repository is tagged.
```

File: README.md [hunk 8]
```diff
@@ -1,23 +1,46 @@
...
+
+Here are the step to follow to push a new image:
```
Comment (jmaupetit) [inline README.md]:
```suggestion
Here are the steps to follow to publish a new image:
```

File: README.md [hunk 9]
```diff
@@ -1,23 +1,46 @@
...
+
+- Edit file `.circleci/releases.sh` and change the tag version corresponding to the image you want to push.
```
Comment (jmaupetit) [inline README.md]:
```suggestion
- Edit the `.circleci/releases.sh` file and update the tag version corresponding to the image you want to publish.
```

File: README.md [hunk 10]
```diff
@@ -1,23 +1,46 @@
...
+- Commit your changes and once merged into master you will be able to tag a new version.
```
Comment (jmaupetit) [inline README.md]:
```suggestion
- Commit your changes, submit a pull request and once merged into master you will be able to tag a new version.
```

File: README.md [hunk 11]
```diff
@@ -1,23 +1,46 @@
...
+
+Tags must follow those pattern depending the image you want to push:
```
Comment (jmaupetit) [inline README.md]:
```suggestion
Tags must stick to the following patterns for the image release you want to publish:
```

File: README.md [hunk 12]
```diff
@@ -1,23 +1,46 @@
...
+
+- `learninglocker-v.*` if you want to push a new learning locker image. _eg_: `learninglocker-v2.6.2`,
```
Comment (jmaupetit) [inline README.md]:
```suggestion
- `learninglocker-[RELEASE_TAG]` if you want to publish a new learning locker image (_e.g._ `learninglocker-v2.6.2`),
```

File: README.md [hunk 13]
```diff
@@ -1,23 +1,46 @@
...
+- `xapi-service-v.*` if you want to push a new xapi-service image. _eg_: `xapi-service-v2.2.15`.
```
Comment (jmaupetit) [inline README.md]:
```suggestion
- `xapi-service-v.[RELEASE_TAG]` if you want to publish a new xapi-service image (_e.g._ `xapi-service-v2.2.15`).
```

File: README.md [hunk 14]
```diff
@@ -1,23 +1,46 @@
...
+
 ## Use learning locker image
```
Comment (jmaupetit) [inline README.md]:
```suggestion
## Using the learning locker image
```

File: README.md [hunk 15]
```diff
@@ -1,23 +1,46 @@
...
 
 You have to configure your learning locker image using environment variables. You will found all available variables in the [project documentation](http://docs.learninglocker.net/guides-configuring/#learning-locker-application).
 
-You also have to use the [xapi-service](https://github.com/LearningLocker/xapi-service) provided by learning locker. They also provide a docker image you can use.
-
 There is no entrypoint nor command configured in this image. You will have to declare them when you run your container. Here are the commands to use:
```
Comment (jmaupetit) [inline README.md]:
```suggestion
There is no entrypoint nor command configured in this image. You will have to declare them explicitly when you run your container. A list of commands to use _per_ service follows:
```

File: README.md [hunk 16]
```diff
@@ -44,6 +67,12 @@ You can also use the CLI by executing the following command from your container:
 $ docker-run --rm fundocker/learninglocker:v2.6.2 node cli/dist/server
 ```
 
+### Use xapi-service image
```
Comment (jmaupetit) [inline README.md]:
```suggestion
### Using the xapi-service image
```

File: README.md [hunk 17]
```diff
@@ -44,6 +67,12 @@ You can also use the CLI by executing the following command from your container:
...
+
+You have to configure your xapi-service image using environment variables. You will found all available variables in the [project documentation](http://docs.learninglocker.net/guides-configuring/#xapi-service).
```
Comment (jmaupetit) [inline README.md]:
```suggestion
You have to configure your xapi-service image using environment variables. You will find all available variables in the [project documentation](http://docs.learninglocker.net/guides-configuring/#xapi-service).
```

File: README.md [hunk 18]
```diff
@@ -1,23 +1,46 @@
...
+- Commit your changes and once merged into master you will be able to tag a new version.
+
+Tags must follow those pattern depending the image you want to push:
+
+- `learninglocker-v.*` if you want to push a new learning locker image. _eg_: `learninglocker-v2.6.2`,
+- `xapi-service-v.*` if you want to push a new xapi-service image. _eg_: `xapi-service-v2.2.15`.
+
 ## Use learning locker image
 
 You have to configure your learning locker image using environment variables. You will found all available variables in the [project documentation](http://docs.learninglocker.net/guides-configuring/#learning-locker-application).
```
Comment (jmaupetit) [inline README.md]:
```suggestion
You have to configure your learning locker image using environment variables. You will find all available variables in the [project documentation](http://docs.learninglocker.net/guides-configuring/#learning-locker-application).
```

File: README.md [hunk 19]
```diff
@@ -44,6 +67,12 @@ You can also use the CLI by executing the following command from your container:
 $ docker-run --rm fundocker/learninglocker:v2.6.2 node cli/dist/server
 ```
 
+### Use xapi-service image
+
+You have to configure your xapi-service image using environment variables. You will found all available variables in the [project documentation](http://docs.learninglocker.net/guides-configuring/#xapi-service).
+
+The entrypoint in this image will run the xapi-service, you have nothing special to do.
```
Comment (jmaupetit) [inline README.md]:
```suggestion
By default, the entrypoint from this image will run the `xapi-service`, so you don't have to declare a command.
```

File: README.md (hunk 15 above)
Comment (lunika) [reply README.md]: weird I don't have the whitespace at the beginning in local...

File: README.md (hunk 16 above)
Comment (lunika) [reply README.md]: I should also change the title for learning locker ? In one hand we have `Using the xapi-service image` and in the other hand `Use learning locker image`

File: README.md (hunk 15 above)
Comment (jmaupetit) [reply README.md]: Have you seen the rest of this suggestion (we don't care about this space)?

File: README.md (hunk 16 above)
Comment (jmaupetit) [reply README.md]: I thought I've added it.

**Comment by lunika** [comment]
@jmaupetit modification made

**Review by jmaupetit — APPROVED** [review]

**Pull request merged**